packages/flutter/test/widgets/text_selection_test.dart

```diff
@@ -380,4 +405,213 @@ void main() {
+    await pumpEditableTextSelectionHandlersProvider(tester);
+    final TestGesture gesture =
+      await tester.startGesture(const Offset(200.0, 200.0), pointer: 0, kind: PointerDeviceKind.touch);
+    await gesture.up();
+    await tester.pumpAndSettle();
+
+    final FakeEditableTextState state = tester.state(find.byType(FakeEditableText));
+    final FakeRenderEditable renderEditable = tester.renderObject(find.byType(FakeEditable));
+    expect(state.showToolbarCalled, isFalse);
+    expect(renderEditable.selectWordEdgeCalled, isTrue);
+  });
+
+  testWidgets('test EditableTextSelectionHandlersProvider double tap', (WidgetTester tester) async {
+    await pumpEditableTextSelectionHandlersProvider(tester);
+    final TestGesture gesture =
+      await tester.startGesture(const Offset(200.0, 200.0), pointer: 0, kind: PointerDeviceKind.touch);
+    await tester.pump(const Duration(milliseconds: 50));
+    await gesture.up();
+    await gesture.down(const Offset(200.0, 200.0));
+    await tester.pump(const Duration(milliseconds: 50));
+    await gesture.up();
+    await tester.pumpAndSettle();
+
+    final FakeEditableTextState state = tester.state(find.byType(FakeEditableText));
+    final FakeRenderEditable renderEditable = tester.renderObject(find.byType(FakeEditable));
+    expect(state.showToolbarCalled, isTrue);
+    expect(renderEditable.selectWordCalled, isTrue);
+  });
+
+  testWidgets('test EditableTextSelectionHandlersProvider forcePress', (WidgetTester tester) async {
+    await pumpEditableTextSelectionHandlersProvider(tester);
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.downWithCustomEvent(
+      const Offset(200.0, 200.0),
+      const PointerDownEvent(
+        pointer: 0,
+        position: Offset(200.0, 200.0),
+        pressure: 3.0,
+        pressureMax: 6.0,
+        pressureMin: 0.0,
+      ),
+    );
+    await gesture.up();
+    await tester.pump();
+
+    final FakeEditableTextState state = tester.state(find.byType(FakeEditableText));
+    final FakeRenderEditable renderEditable = tester.renderObject(find.byType(FakeEditable));
+    expect(state.showToolbarCalled, isFalse);
+    expect(renderEditable.selectWordsInRangeCalled, isTrue);
+  });
+
+  testWidgets('test EditableTextSelectionHandlersProvider selection disabled', (WidgetTester tester) async {
+    await pumpEditableTextSelectionHandlersProvider(tester, selectionEnabled: false);
+    final TestGesture gesture =
+      await tester.startGesture(const Offset(200.0, 200.0), pointer: 0, kind: PointerDeviceKind.touch);
+    await tester.pump(const Duration(seconds: 2));
+    await gesture.up();
+    await tester.pumpAndSettle();
+
+    final FakeEditableTextState state = tester.state(find.byType(FakeEditableText));
+    final FakeRenderEditable renderEditable = tester.renderObject(find.byType(FakeEditable));
+    expect(state.showToolbarCalled, isTrue);
```

Comment:
another place that feels weird to me. Is this also intended? @justinmc

Reply:
Can you help me understand what this is testing?  Definitely long pressing on an unfocused text field should show the selection menu.  I'm not sure if that's exactly what's being tested though, like how `selectionEnabled: false` affects this.

Reply:
Does `showToolbarCalled` mean that the toolbar actually would be shown, or could it be blocked when selection is disabled?  Same question for `selectWordsInRangeCalled`.

Reply:
selectionEnabled false behave the same as EditableText.selectionEnabled.
showToolbarCalled means the it try to show the toolbar, and in the real case, that would have shown the toolbar.

I am trying to test when selectionEnabled is false, Long press should not try to call the selectWordsInRange, and to my surprise it still try to show the toolbar.

Reply:
Oh got it, thank you.  I think this is a bug that I have a PR out for!  This is the issue: https://github.com/flutter/flutter/issues/34215.  So after my PR is merged this test will need to be changed if I'm right.

Reply:
It is still passing. The issue is onSingleLongTapEnd is not guarded by selectionEnabled. I can easily fix this in my code, but i am not sure which one is the desired behavior

Reply:
I will keep it as is, this since not related to this pr, if we want to fix this(not sure if we want to), we should create a new issue

Reply:
Sounds good.  Since everything is still passing I think all is well.